tests/viewsets.py

```diff
@@ -103,6 +104,27 @@ class LocationViewSet(DynamicModelViewSet):
     queryset = Location.objects.all()
 
 
+class AlternateLocationViewSet(DynamicModelViewSet):
+    model = Location
+    serializer_class = LocationSerializer
+    queryset = Location.objects.all()
+
+    def filter_queryset(self, queryset):
+        user_name = self.request.query_params.get('user_name')
+        user_name_separate_filter = self.request.query_params.get(
+            'user_name_separate',
+        )
+        if user_name:
+            self.extra_drest_filters = Q(
```

Comment:
Hm... I didn't realize you intended this to be dynamic. An attribute technically works, but given that the other DREST attributes are effectively constants, it would be more consistent to use a method: maybe something like `def get_extra_filters(self, request)`.